### ISSUE TITLE
What is the connection between the parameter l2_leaf_reg and the function in this website?

### ISSUE THREAD

**Issue by Wangpc-972** [catboost/catboost#1757]
My code is as follows：

```
model = CatBoostClassifier(iterations=100,
                            learning_rate=0.3,
                            l2_leaf_reg=0.12,
                            custom_loss=['AUC','Accuracy'],
                            score_function='L2')
```
I want to ask what is the relationship between the parameter `l2_leaf_reg` and the parameter `score_function`？In [here](https://catboost.ai/docs/concepts/algorithm-score-functions.html), the l2 regularization parameter is mentioned in the Second-order score functions section，is it the parameter `l2_leaf_reg` that I set in the code, and how is the final score function of Catboost specifically obtained?

**Comment by ek-ak** [comment]
Hello! Yes, it is that `l2_leaf_reg`, you are right. The final score function formula is [here](https://catboost.ai/docs/concepts/algorithm-score-functions.html#per-object-feature-penalties), and L2 score function formula is in the first half of [Finding the optimal tree structure](https://catboost.ai/docs/concepts/algorithm-score-functions.html#optimal-tree-structure) section.

**Comment by Wangpc-972** [comment]
> Hello! Yes, it is that `l2_leaf_reg`, you are right. The final score function formula is [here](https://catboost.ai/docs/concepts/algorithm-score-functions.html#per-object-feature-penalties), and L2 score function formula is in the first half of [Finding the optimal tree structure](https://catboost.ai/docs/concepts/algorithm-score-functions.html#optimal-tree-structure) section.

Thank you for your reply！

What is the relationship between the final score function formula and the L2 score function formula? If I set `score_function='L2'`, is the final score function used by Catboost the L2 score function formula?

**Comment by kizill** [comment]
No, l2_leaf_reg only affects binary splits scoring and leaf values calculation and do not affect loss function evaluation and eval metrics calculation after tree structure is selected and leaf values calculated. Feel free to reopen if you have some questions! 😺

**Comment by Wangpc-972** [comment]
Thank you,
      I am sorry to see your reply recently. I also want to know if the regularization parameter is too large (or too small), will CatBoost overfit or underfit?
